guides/pages/how-to-synchronize-your-yjs-storage-data-to-a-planetscale-mysql-database.mdx

```diff
@@ -0,0 +1,584 @@
+back after you’re set up.
+
+</Banner>
+
+### This specific webhook is throttled
+
+Note that the [YDocUpdated](/docs/platform/webhooks#YDocUpdatedEvent) webhook
+event is throttled at a rate of _once every 5 minutes_. This is because Yjs can
+update up to _60 times per second_, and it would be impractical to run the
+webhook this frequently.
+
+## Create an endpoint in your project
+
+When a webhook event is triggered, it can send POST request to the back end in
+your project, and from within there we can update the database. In this guide,
+we’ll be using a Next.js route handler (API endpoint) as an example, but other
+frameworks work similarly.
+
+In order to use webhooks, we’ll need to retrieve the `headers` and `body` from
+the request. Here’s the basic endpoint we’ll be starting from:
+
+```ts
+export async function POST(request: Request) {
+  const body = await request.json();
+  const headers = request.headers;
+
+  // Handle webhooks and database
+  // ...
+
+  return new Response(null, { status: 200 });
+}
+```
+
+Create this endpoint in your project, and make it available on `localhost` at
+the following URL:
+
+```
+/api/liveblocks-database-sync
+```
+
+### Testing webhooks locally
+
+Running webhooks locally can be difficult, but one way to do this is to use a
+tool such as [`localtunnel`](https://www.npmjs.com/package/localtunnel) or
+[`ngrok`](https://www.npmjs.com/package/ngrok) which allow you to temporarily
+put your localhost server online.
+
+If your project is running on `localhost:3000`, you can run the following
+command to generate a temporary URL that’s available while your localhost server
+is running:
+
+```bash
+npx localtunnel --port 3000
+```
+
+If you correctly input your IP address, the URL `localtunnel` generates can be
+placed into the Liveblocks webhooks dashboard for quick testing.
+
+## Set up webhooks on the Liveblocks dashboard
+
+To use webhooks, you need to add your endpoint URL to the webhooks dashboard
+inside your Liveblocks project, and tell the webhook to trigger when Yjs
+document data has changed.
+
+<Steps>
+  <Step>
+    <StepTitle>Select your project</StepTitle>
+    <StepContent>
+      From the [Liveblocks dashboard](/dashboard), navigate to the project you’d
+      like to use with webhooks, or create a new project.
+      
+      <Figure>
+        <Image
+          src="/assets/tutorials/webhooks/liveblocks-create-project.png"
+          alt="Create a Liveblocks project"
+          width={969}
+          height={647}
+        />
+      </Figure>
+    </StepContent>
+  </Step>
+
+{" "}
+
+<Step>
+  <StepTitle>Go to the webhooks dashboard</StepTitle>
+  <StepContent>
+    Click on the “Webhooks” tab on the menu at the left.
+    <Figure>
+      <Image
+        src="/assets/tutorials/webhooks/liveblocks-click-webhooks.png"
+        alt="Click webhooks"
+        width={1150}
+        height={602}
+      />
+    </Figure>
+  </StepContent>
+</Step>
+
+{" "}
```

Comment:
What does this do? I only looked at the code in GitHub, not tested linking this branch to see the actual result.

Reply:
> What does this do? I only looked at the code in GitHub, not tested linking this branch to see the actual result.

Prettier keeps adding it here, and I have no idea why! it doesn't do anything visible